server/src/main/java/org/opensearch/action/search/TransportCreatePITAction.java

```diff
@@ -0,0 +1,341 @@
+                            searchPhaseResultActionListener.onResponse(response);
+                        }
+
+                        @Override
+                        public void handleException(TransportException exp) {
+                            searchPhaseResultActionListener.onFailure(exp);
+                        }
+
+                        @Override
+                        public String executor() {
+                            return ThreadPool.Names.GENERIC;
+                        }
+                    }
+                );
+            }
+        }, createPitListener);
+
+        /**
+         * Phase 2 of create PIT : Update PIT reader context with PIT ID and keep alive from request
+         * Fail create pit operation if any of the updates in this phase are failed
+         */
+        createPitListener.whenComplete(searchResponse -> {
+            SearchContextId contextId = SearchContextId.decode(namedWriteableRegistry, searchResponse.pointInTimeId());
+            final StepListener<BiFunction<String, String, DiscoveryNode>> lookupListener = getConnectionLookupListener(contextId);
+            lookupListener.whenComplete(nodelookup -> {
+                final ActionListener<TransportCreatePITAction.UpdatePitContextResponse> groupedActionListener = getGroupedListener(
+                    updatePitIdListener,
+                    searchResponse,
+                    contextId.shards().size()
+                );
+                /**
+                 * store the create time ( same create time for all PIT contexts across shards ) to be used
+                 * for list PIT api
+                 */
+                TimeValue createTime = new TimeValue(System.currentTimeMillis());
+                for (Map.Entry<ShardId, SearchContextIdForNode> entry : contextId.shards().entrySet()) {
+                    DiscoveryNode node = nodelookup.apply(entry.getValue().getClusterAlias(), entry.getValue().getNode());
+                    try {
+                        final Transport.Connection connection = searchTransportService.getConnection(
+                            entry.getValue().getClusterAlias(),
+                            node
+                        );
+                        searchTransportService.updatePitContext(
+                            connection,
+                            new UpdatePITContextRequest(
+                                entry.getValue().getSearchContextId(),
+                                searchResponse.pointInTimeId(),
+                                request.getKeepAlive().millis(),
+                                createTime
+                            ),
+                            groupedActionListener
+                        );
+                    } catch (Exception e) {
+                        groupedActionListener.onFailure(e);
+                    }
+                }
+            }, updatePitIdListener::onFailure);
+        }, updatePitIdListener::onFailure);
```

Comment:
This is incomplete. If 5 out of 10 PIT contexts are updated then we need to delete the PIT. Let's delete the PIT. Otherwise we will have zombie pit contexts which aren't accessible.

Reply:
We'll add delete once delete pit is done